src/InstagramClone.Api/Database/InstagramCloneDbContext.cs

```diff
@@ -36,13 +37,23 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
                 .HasOne(following => following.Follower)
                 .WithMany(followers => followers.Users);
 
+
             modelBuilder.Entity<UserFollower>()
-                .HasOne(f => f.Follower)
+                .HasOne<User>(us => us.FollowedUser)
                 .WithMany(u => u.Followed);
 
             modelBuilder.Entity<UserFollower>()
-                .HasOne(fl => fl.FollowedUser)
-                .WithMany( w => w.Followers);
+                .HasOne(d => d.FollowedUser)
+                .WithMany(m => m.Followers)
```

Comment:
did you check this too? Think about the difference between follower and Followed.

why are you matching follower to followed?